odl/solvers/functional/functional.py

```diff
@@ -0,0 +1,1160 @@
+        example `forward_backward_pd`.
+        """
+        raise NotImplementedError
+
+    @property
+    def convex_conj(self):
+        """Convex conjugate functional of the functional.
+
+        Notes
+        -----
+        The convex conjugate functional of a convex functional :math:`f(x)`,
+        defined on a Hilber space, is defined as the functional
+
+        .. math::
+
+            f^*(x^*) = \\sup_{x} \{ \\langle x^*,x \\rangle - f(x)  \}.
+
+        The concept is also known as the Legendre transformation.
+
+        References
+        ----------
+        Wikipedia article on `Convex conjugate
+        <https://en.wikipedia.org/wiki/Convex_conjugate>`_.
+
+        Wikipedia article on `Legendre transformation
+        <https://en.wikipedia.org/wiki/Legendre_transformation>`_.
+
+        For literature references see, e.g., [Lue1969]_, [Roc1970]_.
+        """
+        raise NotImplementedError
+
+    def derivative(self, point):
+        """Return the derivative operator in the given point.
+
+        This function returns the linear operator given by
+
+            ``self.derivative(point)(x) == self.gradient(point).inner(x)``
+
+        Parameters
+        ----------
+        point : `LinearSpaceElement`
+            The point in which the gradient is evaluated.
+
+        Returns
+        -------
+        out : `Operator`
+        """
+        return self.gradient(point).T
+
+    def translated(self, shift):
+        """Return a translation of the functional.
+
+        For a given functional ``f`` and an element ``translation`` in the
+        domain of ``f``, this operation creates the functional
+        ``f(. - translation)``.
+
+        Parameters
+        ----------
+        translation : `LinearSpaceElement`
+            Element in the domain of the functional
+
+        Returns
+        -------
+        out : `TranslatedFunctional`
+            The functional ``f(. - translation)``
+        """
+        if shift not in self.domain:
+            raise TypeError('`shift` {} is not in the domain of the '
+                            'functional {!r}'.format(shift, self))
+
+        return TranslatedFunctional(self, shift)
+
+    def __mul__(self, other):
+        """Return ``self * other``.
+
+        If ``other`` is an `Operator`, this corresponds to composition with the
+        operator:
+
+            ``func * op == (x --> func(op(x)))``
+
+        If ``other`` is a scalar, this corresponds to right multiplication of
+        scalars with functionals:
+
+            ``func * scalar == (x --> func(scalar * x))``
+
+        If ``other`` is a vector, this corresponds to right multiplication of
+        vectors with functionals:
+
+            ``func * vector == (x --> func(vector * x))``
+
+        Note that left and right multiplications are generally different.
+
+        Parameters
+        ----------
+        other : `Operator`, `LinearSpaceElement` or scalar
```

Comment:
`LinearSpaceElement` -> `domain` element.

Apply this change throughout this file.